src/components/SortControl.tsx

```diff
@@ -0,0 +1,26 @@
+import React, { useState, useEffect } from 'react';
+
+interface SortControlProps {
+    currentSelection: string;
+    onSortChange: (newSelection: string) => void;
+}
+
+const SortControl: React.FC<SortControlProps> = ({ currentSelection, onSortChange }) => {
+    const [selectedOption, setSelectedOption] = useState(currentSelection);
+
+    useEffect(() => {
+        onSortChange(selectedOption);
+    }, [selectedOption, onSortChange]);
```

Comment:
I don't see why this components needs to implement these internal piece of state and use effect. It should just take `currentSelection`, render it and call `onSortChange` with a new value.